liri.js

```diff
@@ -0,0 +1,195 @@
+
+		for (var i=0; i< tweets.statuses.length; i++) {
+			logStr = logStr+tweets.statuses[i].created_at+'\n';
+			logStr = logStr+tweets.statuses[i].text+'\n';
+		}
+		console.log(logStr);
+		logOutput(logStr);
+	});
+}
+
+function spotifyThisSong(song) {
+	var spotifyKeys = keys.spotifyKeys;
+	var spotify = new Spotify({
+  		id: spotifyKeys.client_id,
+	  	secret: spotifyKeys.client_secret
+	});
+
+	var params = {
+		type: 'track', 
+		query: "\""+song+"\"", 
+		limit: '1'
+	}
+
+	spotify.search(params, function(err, data) {
+	  	if (err) {
+	    	console.log('Error occurred: ' + err);
+	    	logOutput('Error occurred: ' + err + '\n');
+	    	return;
+	  	}
+
+		for (var i = 0; i<data.tracks.items.length; i++) {
+			var logStr = ""; // Save the output to this string and display it on console and append to log.txt
+
+			var songItem = data.tracks.items[i];
+			logStr = logStr+"Song: " + songItem.name+'\n';
+			logStr = logStr+"Album: " + songItem.album.name+'\n';
+			var len = songItem.album.artists.length;
+			var artists = "";
+			for (var j=0; j < len; j++) {
+				artists = artists + songItem.album.artists[j].name + ' ';
+			}
+			logStr = logStr+"Artist(s): " + artists.trim()+'\n';
+			logStr = logStr+"Preview Link: " + songItem.preview_url+'\n';
+			
+			console.log(logStr);
+			logOutput(logStr);
+		}
+	});
+}
+
+function movieThis(movie) {
+	var accessKey = keys.oMDbAccessKey.accessKey;
+	var queryURL = "http://www.omdbapi.com/?t=" + movie + "&y=&plot=short&apikey=" + accessKey;
+	request(queryURL, function(err, response, body){
+		if (err) {
+			console.log("OMDB Error: " + err);
+			logOutput("OMDB Error: " + err+'\n');
+			return;
+		}
+		if (response.statusCode != 200) {
+			console.log("Failed - OMDb Response Code: " + response.statusCode);
+			logOutput("Failed - OMDb Response Code: " + response.statusCode+'\n');
+			return;
+		}
+
+		var logStr = ""; // Save the output to this string and display it on console and append to log.txt
+		var result = JSON.parse(body);
+		logStr = logStr+"Title: " + result.Title+'\n';
+		logStr = logStr+"Year: " + result.Year+'\n';
+
+		if (result.Ratings.length == 0) {
+			logStr = logStr+"Internet Movie Database Rating: Not Available"+'\n';
+			logStr = logStr+"Rotten Tomatoes Rating: Not Available"+'\n';
+		}
+
+		if (result.Ratings.length == 1) {
+			logStr = logStr+"Internet Movie Database Rating: " + result.Ratings[0].Value+'\n';
+			logStr = logStr+"Rotten Tomatoes Rating: Not Available"+'\n';	
+		}
+
+		if (result.Ratings.length == 2) {
+			logStr = logStr+"Internet Movie Database Rating: " + result.Ratings[0].Value+'\n';
+			logStr = logStr+"Rotten Tomatoes Rating: " + result.Ratings[1].Value+'\n';
+		}
+		
+		logStr = logStr+"Country: " + result.Country+'\n';
+		logStr = logStr+"Language: " + result.Language+'\n';
+		logStr = logStr+"Plot: " + result.Plot+'\n';
+		logStr = logStr+"Actors: " + result.Actors+'\n';
+		console.log(logStr);
+		logOutput(logStr);
+	});
```

Comment:
This code is not very scalable and adaptable to the possible different results you could get. instead, I would recommend an iterative function that just takes the ratings and creates one string to be logged.